numpy/ma/core.py

```diff
@@ -5108,77 +5043,143 @@ def anom(self, axis=None, dtype=None):
+        x = asanyarray(self - arrmean)
+        if issubclass(self.dtype.type, np.complexfloating):
+            x = np.multiply(x, np.conjugate(x), out=x).real
+        else:
+            x = np.multiply(x, x, out=x)
+        ret = x.sum(axis, dtype, **kwargs)
+        if ret is masked:
+            return _return_masked(out)
+
+        # Compute degrees of freedom and make sure it is not negative.
+        rcount = self.count(axis=axis, **kwargs)
+        rcount = (rcount - ddof).clip(min=0)
+
+        # divide by degrees of freedom
+        with np.errstate(divide='ignore', invalid='ignore'):
+            if isinstance(ret, np.ndarray):
+                ret = true_divide(
+                        ret, rcount, out=ret, casting='unsafe', subok=False)
+            elif hasattr(ret, 'dtype'):
+                ret = ret.dtype.type(ret / rcount)
+            else:
+                ret = ret / rcount
+
+        # account for case we got a scalar from sum, and rcout == 0
+        if ret.ndim == 0 and np.isnan(ret):
+            ret = masked
+
         if out is not None:
-            # Set the data
-            out.flat = dvar
-            # Set the mask if needed
             if isinstance(out, MaskedArray):
-                out.__setmask__(dvar.mask)
+                out.flat = ret
+                outmask = getattr(out, '_mask', nomask)
+                if (outmask is nomask):
+                    outmask = out._mask = make_mask_none(out.shape)
+                outmask.flat = getattr(ret, '_mask', nomask)
+            else:
+                if out.dtype.kind in 'biu' and any(~ret._mask):
+                    errmsg = "Masked data information would be lost in one or "\
+                             "more location."
+                    raise MaskError(errmsg)
+                out.flat = ret.filled(np.nan)
             return out
-        return dvar
-    var.__doc__ = np.var.__doc__
 
-    def std(self, axis=None, dtype=None, out=None, ddof=0):
-        ""
-        dvar = self.var(axis=axis, dtype=dtype, out=out, ddof=ddof)
+        return ret
+
+    def std(self, axis=None, dtype=None, out=None, ddof=0, **kwargs):
+        """
+        Returns the standard deviation of the array elements along given axis.
+
+        Masked entries are ignored.
+
+        Refer to `numpy.std` for full documentation.
+
+        See Also
+        --------
+        ndarray.std : corresponding function for ndarrays
+        numpy.std : Equivalent function
+        """
+        dvar = self.var(axis, dtype, out, ddof, **kwargs)
         if dvar is not masked:
             if out is not None:
                 np.power(out, 0.5, out=out, casting='unsafe')
                 return out
             dvar = sqrt(dvar)
         return dvar
-    std.__doc__ = np.std.__doc__
 
     def round(self, decimals=0, out=None):
         """
-        Return an array rounded a to the given number of decimals.
+        Return `a` with each element rounded to the given number of decimals.
```

Comment:
What is `a`? You can probably just remove it maybe: Return each element rounded or so. I see its copied from `ndarray.round` doesn't make it better. (there the signature is "a.round(...), which makes it half right)